src/functions/Output.ps1

```diff
@@ -724,6 +805,10 @@ function Get-WriteScreenPlugin ($Verbosity) {
                 if ($PesterPreference.Output.Verbosity.Value -in 'Detailed', 'Diagnostic') {
                     & $SafeCommands['Write-Host'] -ForegroundColor $ReportTheme.Skipped "$margin[!] $out" -NoNewLine
                     & $SafeCommands['Write-Host'] -ForegroundColor $ReportTheme.SkippedTime " $humanTime"
+
+                    if ($PesterPreference.Run.SkipRemainingOnFailure.Value -in 'Block', 'Container', 'Run') {
+                        & $SafeCommands['Write-Host'] -ForegroundColor $ReportTheme.Skipped "$margin Skipped due to previous failure and Run.SkipRemainingOnFailure set to '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
+                    }
```

Comment:
Will print message (false positive) when users use `-Skip` and plugin is active.  Ex. `It 'test1' -Skip { }`. No way to split plugin-caused skip from user-caused skip.

Only `Set-ItResult` stores the reason why skip was set by adding a PesterTestSkipped-error record to the test-object. Ex.

```powershell
$test.ErrorRecord.Add([Pester.Factory]::CreateErrorRecord( 
     'PesterTestSkipped', #string errorId 
     $Message, #string message 
     $File, #string file 
     $Line, #string line 
     $LineText, #string lineText 
     $false #bool terminating 
 ))
```

Should we drop the printed output and add a similar error-record to the tests being marked to skip? Could also mention the failing test that caused it.

Reply:
Yeah I think dropping printed output would be a good idea here, I completely forgot about `-Skip` for tests. Adding to error record seems like a better idea, which could contain the failing test inside `Message`.

Reply:
I can probably create another function to set add this error record, with an errorID of something like `pesterRemainingTestSkipped`? Just to differentiate the fully qualified error id from the one `Set-ItResult` sets.

Reply:
@fflaten Would you be happy with something like the following being added:

```pwsh
$failingTestPath = $Context.Test.ExpandedPath
$failingTestLine = $Context.Test.ScriptBlock.StartPosition.StartLine

$errorRecord = [Pester.Factory]::CreateErrorRecord(
    'PesterRemainingTestSkipped',
    "Skipped due to previous failure ($($failingTestPath):$($failingTestLine)) and Run.SkipRemainingOnFailure set to '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'",
    $test.ScriptBlock.File,
    $test.ScriptBlock.StartPosition.StartLine,
    $test.ExpandedPath,
    $false
)
$test.ErrorRecord.Add($errorRecord)
```

This would add the failing test path and number to the message, and File/Line/LineText would just contain the info of the current skipped test.

Reply:
Not sure if errorid matters, but I'd probably use `PesterTestSkipped` to be consistent. This is a user-controlled skip just like `Set-ItResult`, only with a different trigger. That reason is shown in the error-message.

The error-record should include `$Context.Test.ExpandedPath` which might help identify the testcase that failed if they used variables in the test or block-names. I don't think File, StartPosition and Scriptblock provides any value in the record since they belong to tests that hasn't been executed. If not null, they could point at the file/startpos/line for the original error that caused the failed test, but that might be bad practice (referencing different code in a error). Thoughts @nohwnd ?

Btw. `$test.StartLine` is preferred over `$test.ScriptBlock.Startline` as the latter will return the line of `{` which might be multiple lines down when using testcases. `$test.StartLine` should always return the line of `It ...`

Reply:
> Not sure if errorid matters, but I'd probably use PesterTestSkipped to be consistent. This is a user-controlled skip just like Set-ItResult, only with a different trigger. That reason is shown in the error-message.

Yeah I've kept it as `PesterTestSkipped`, which now seems to also show the skipped message next to the test inside my update PR description screenshots. 

> $Context.Test.ExpandedPath might be a better reference to the failed test as it can help identity a specific testcase (if variables are used in name). I don't think File, StartPosition and Scriptblock provides any value in the record since they belong to tests that hasn't been executed. If not null, they could point at the file/startpos/line for the original error that caused the failed test, but that might be bad practice (referencing different code in a error). Thoughts @nohwnd ?

Also removed File,StartPosition and Scriptblock from the error record and populating with `$null` for now. Happy to put in more suitable values here if needed. 

> Should it output a single line at the end of the run in Detailed+Diagnostic to mention that one or more tests were skipped due to Run.Skip... ? Or is it unnecessary noise considering it's a explicit opt-in behavior?

I feel like that since this will be opt-in behavior, the user would probably want as little noise as possible. I'm still wondering what we should do with the skipped message that is added from `PesterTestSkipped` error record being added. I feel like if a 100 tests are running this is going to get really noisy real fast, making me think a message at the bottom for Detailed/Diagnostic is probably the most feasible option.

Reply:
> I also see that this message is always added to `PesterTestSkipped` error records: https://github.com/ArmaanMcleod/Pester/blob/main/src/functions/Output.ps1#L645-L653. I think its really useful to add this error record, especially when using `PassThru`. 

Is that actually executed? Will have to test later, but I wouldn't expect it since it's part of test teardown. The first test is marked as failed, the rest are marked to be skipped which shouldn't invoke any setup, test or teardown for them.

Reply:
You're right. TIL the plugin steps are are actually called even though the setup/teardowns/tests/blocks themselves will be skipped after. Makes sense considering the steps are called ex. EachTestSetup**Start**, EachTestTeardown**End** etc., but still not sure if it's intentional or just a result of calling EachTestSetupStart using the same `Invoke-ScriptBlock` as OneTimeTestSetupStart.

It's easy enough to exclude with `if(-not $_test.Skip -and $_test.ErrorRecord.FullyQualifiedErrorId -eq 'PesterTestSkipped') { ...` in the Output-plugin - or use a different errorid ofc. I'm curious to know what @nohwnd thinks about this. Currently:
- `It 't1' -Skip` will only output `[!] t1`
- `Set-ItResult` will include a message/reason
- What should this do? Since it's opt-in and "planned" like `-Skip`, then ignoring the message might be the enough. This also aligns with the recent change to reduce noise in Detailed and Diagnostic. Optionally include a single message somewhere (RunEnd, ContainerEnd, BlockEnd depending on option? Or just at the end?)

Reply:
I think excluding is probably easiest for now. I can also add a message to Detailed/Diagnostic output at the end if needed. Up to you guys 🙂 .

Reply:
> but still not sure if it's intentional or just a result of calling EachTestSetupStart using the same Invoke-ScriptBlock as OneTimeTestSetupStart
@fflaten  do you mean this part? That runs the script in two different scopes. I don't think I intentionally did it this way. But imho it is correct. The  Skipped state is just a short circuiting the execution of the test.

https://github.com/pester/Pester/blob/main/src/Pester.Runtime.ps1#L567-L579

Reply:
> @fflaten  do you mean this part? That runs the script in two different scopes. I don't think I intentionally did it this way. But imho it is correct. The  Skipped state is just a short circuiting the execution of the test.
> 
> https://github.com/pester/Pester/blob/main/src/Pester.Runtime.ps1#L567-L579

Yeah, EachTestSetupStart + EachTestTeardownEnd are invoked even though BeforeEach/AfterEach won't be executed (due to skip-detection in the next lines). Nothing we need to solve here, it was just unexpected for someone unexperienced with the plugin-system. 👍

I'll probably post a few issues with some questions + demo around this later considering the plugin-system will be made public at some point. As part of that initiative the code and docs needs to be clear about when to expect which step to be executed. IIRC there's even some steps that's currently not invoked at all, including RunEnd that I suggested above. 🙂